node-src/git/getChangedFilesWithReplacement.ts

```diff
@@ -39,20 +39,41 @@ export async function getChangedFilesWithReplacement(
       `Got error fetching commit for #${build.number}(${build.commit}): ${err.message}`
     );
 
-    if (/(bad object|uncommitted changes)/.test(err.message)) {
-      const replacementBuild = await findAncestorBuildWithCommit(ctx, build.number);
+    if (!/(bad object|uncommitted changes)/.test(err.message)) {
+      throw err;
+    }
+
+    // Try to find a replacement build by checking multiple ancestor builds
+    let skip = 0;
```

Comment:
This loop is essentially what `findAncestorBuildWithCommit` does. Should `findAncestorBuildWithCommit` be responsible for returning a valid build? Again, it seems like it does but the `git` command fails. I'm just not sure trying another ancestor build would yield better results.